### PR TITLE
#167735030 Setup Docker for the backend application

### PR DESCRIPTION
**Pull request by elikeyz**
#### What does this PR do?
Setup Docker to build an image to run the application
#### Description of Task to be completed?
- Create a Dockerfile to write instructions for Docker to create the image
- Create a .dockerignore file to exclude certain files and folders from being added to the image
- Write instructions for running and building the Docker image in Readme.MD
#### How should this be manually tested?
See the `Installation and Usage with Docker` section on the Readme for instructions
#### Any background context you want to provide?
Presently, the Docker image built has the port set to the development port. The production port is not yet accounted for at this time.
#### What are the relevant pivotal tracker stories?
[#167735030](https://www.pivotaltracker.com/n/projects/2382051)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+RUN npm install
+
+# Bundle app source code inside the Docker image
+COPY . .
+
+# Map port to the Docker daemon
+EXPOSE 3000
+
+# Run the start script
+CMD [ "npm", "start" ]
```
Comment (Annmary12) [inline Dockerfile]: Leave a space after line 21

File: README.md [hunk 2]
```diff
@@ -1,7 +1,25 @@
-Barefoot Nomad - Making company travel and accomodation easy and convinient.
+Barefoot Nomad - Making company travel and accomodation easy and convenient.
 =======
 
 ## Vision
 Make company global travel and accommodation easy and convenient for the strong workforce of savvy members of staff, by leveraging the modern web.
 
 ---
+
+## Installation and Usage with Docker
```
Comment (Annmary12) [inline README.md:9]: Good work @elikeyz on setting up the docker file.
Things you need to note:
- You can add a link on how to setup docker on the system or short description on how to set it up on the readme.
- Do you necessarily need to add your username? remember this project is a team's project, not a personal project. leaving it this way `docker build -t iceman-backend` is fine

File: Dockerfile (hunk 1 above)
Comment (elikeyz) [reply Dockerfile]: I've added the empty line

File: README.md (hunk 2 above)
Comment (elikeyz) [reply README.md:9]: I've made the changes

**Comment by Annmary12** [comment]
@elikeyz , kindly resolve the conflict on your PR

**Pull request merged**